### ISSUE TITLE
missing_inline_in_public_items: false positive on externally exported function

### ISSUE THREAD

**Issue by DaniPopes** [rust-lang/rust-clippy#15301]
### Summary

Adding `#[inline]` to an externally exported function triggers `unused_attributes`, so `missing_inline_in_public_items` shouldn't trigger.

### Lint Name

missing_inline_in_public_items

### Reproducer

I tried this code:

```rust
#![warn(clippy::missing_inline_in_public_items)]

#[no_mangle]
pub extern "C" fn my_fn() {}
```

Adding inline gives:

```
warning: `#[inline]` is ignored on externally exported functions
  --> src/lib.rs:91:1
   |
91 | #[inline]
   | ^^^^^^^^^
   |
   = help: externally exported functions are functions with `#[no_mangle]`, `#[export_name]`, or `#[linkage]`
   = note: `#[warn(unused_attributes)]` on by default
```

I expected to see this happen: no fp

### Version

```text
rustc 1.90.0-nightly (5795086bd 2025-07-16)
binary: rustc
commit-hash: 5795086bdfe7ed988aa53a110bd0692c33d8755b
commit-date: 2025-07-16
host: x86_64-unknown-linux-gnu
release: 1.90.0-nightly
LLVM version: 20.1.8
```

**Comment by profetia** [comment]
Have you checked the nightly version of Clippy? I cannot reproduce on nightly.

**Comment by DaniPopes** [comment]
Yes, you need a lib.rs

**Comment by y21** [comment]
Is the reproducer incomplete or am I missing where this function is externally exported? Just specifying "extern" doesn't seem to be enough to get this warning to trigger? I'd assume there's also a `#[no_mangle]`/`#[export_name]`/`#[linkage]` needed for this to reproduce?
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=7d6e3e1bb5e36eccf01c455d6a332ff0

**Comment by DaniPopes** [comment]
oh yeah sorry I missed the no_mangle in the example code
